algorithms/src/snark/marlin/ahp/prover/state.rs

```diff
@@ -68,6 +68,22 @@ pub struct State<'a, F: PrimeField, MM: MarlinMode> {
     /// The length of this list must be equal to the batch size.
     pub(super) z_b: Option<Vec<Vec<F>>>,
 
+    /// The list of Cz vectors for each instance in the batch.
+    /// The length of this list must be equal to the batch size.
+    pub(super) z_c: Option<Vec<Vec<F>>>,
+
+    /// The list of multiplication selector vectors for each instance in the batch.
+    /// The length of this list must be equal to the batch size.
+    pub(super) s_m: Vec<Vec<F>>,
+
+    /// The list of lookup selector vectors for each instance in the batch.
+    /// The length of this list must be equal to the batch size.
+    pub(super) s_l: Vec<Vec<F>>,
```

Comment:
Since we support only the same circuit for all entries in a batch, this is shared across the entire batch, and should live only in the index.